lib/runtime/storage/trie.go

```diff
@@ -17,97 +17,129 @@ import (
 // TrieState is a wrapper around a transient trie that is used during the course of executing some runtime call.
 // If the execution of the call is successful, the trie will be saved in the StorageState.
 type TrieState struct {
```

Comment:
Maybe, could be helpful to have a maximum depth of transactions to prevent the memory of growing up indefinitely

Reply:
I would say if substrate/kagome bound in this way could be good to do, else probably okay to leave it as is. I'm not sure if we want an arbitrary limit here, but a non arbitrary one could be nice

Reply:
Yeah, I didn't find a limit value on spec or substrate code, so I will leave as it is.

Reply:
Agree! thanks for the discussion, we can mark this conversation as resolved.